include/vcpkg/base/contractual-constants.h

```diff
@@ -332,6 +332,7 @@ namespace vcpkg
     inline constexpr StringLiteral FileBaselineDotJson = "baseline.json";
     inline constexpr StringLiteral FileBin = "bin";
     inline constexpr StringLiteral FileControl = "CONTROL";
+    inline constexpr StringLiteral FileCompilerFileHashCacheDotJson = "compiler-file-hash-cache.json";
```

Comment:
I didn't put it here initially because you could put the file anywhere and wouldn't break anyone. I don't know what the exact rule is (when to put stuff in this header and when not).

Reply:
I think the idea is that any 'magic names' should go in here. This thing mostly exists to prevent typoing the name should we need to reference it somewhere else again, at least that's my understanding based on @ras0219-msft 's feedback that created this header in the first place.